curvefs/test/mds/chunkid_allocator_test.cpp

```diff
@@ -0,0 +1,203 @@
+namespace mds {
+
+using curve::kvstorage::KVStorageClient;
+using std::function;
+using std::make_shared;
+using std::set;
+using std::shared_ptr;
+using std::string;
+using std::vector;
+using ::testing::_;
+using ::testing::AtLeast;
+using ::testing::DoAll;
+using ::testing::Invoke;
+using ::testing::Return;
+using ::testing::ReturnArg;
+using ::testing::SaveArg;
+using ::testing::SetArgPointee;
+using ::testing::SetArgReferee;
+using ::testing::StrEq;
+
+uint64_t CHUNKID = 0;  // use to record chunkId
+
+// replace MockEtcdClientImpl::Get
+function<int(const std::string&, std::string*)> GetRep =
+    [&CHUNKID](const std::string& key, std::string* out) -> int {
+    *out = ChunkIdAllocatorImpl::EncodeID(CHUNKID);
+    LOG(INFO) << "check chunk id is: " << *out;
+    return 0;
+};
+
+// replaceMockEtcdClientImpl::CompareAndSwap
+function<int(const std::string&, const std::string&, const std::string&)>
+    CompareAndSwapRep = [&CHUNKID](const std::string& key,
+                                   const std::string& preV,
+                                   const std::string& target) -> int {
+    uint64_t preValue = 0;
+    ChunkIdAllocatorImpl::DecodeID(preV, &preValue);
+    uint64_t targetValue = 0;
+    ChunkIdAllocatorImpl::DecodeID(target, &targetValue);
+    if (preValue != CHUNKID) {
+        LOG(ERROR) << "Error："
+                   << "prev:" << preValue << ", now: " << CHUNKID
+                   << ", target: " << target;
+        return EtcdErrCode::EtcdOutOfRange;
+    }
+
+    CHUNKID = targetValue;
+    return EtcdErrCode::EtcdOK;
+};
+
+class ChunkIdAllocatorTest : public testing::Test {
+ protected:
+    void SetUp() override;
+    void TearDown() override;
+
+    std::shared_ptr<ChunkIdAllocator> chunkIdAllocator_;
+    std::shared_ptr<MockEtcdClientImpl> mockEtcdClient_;
+};
+
+void ChunkIdAllocatorTest::SetUp() {
+    mockEtcdClient_ = make_shared<MockEtcdClientImpl>();
+    chunkIdAllocator_ = make_shared<ChunkIdAllocatorImpl>(mockEtcdClient_);
+}
+
+void ChunkIdAllocatorTest::TearDown() {}
+
+TEST_F(ChunkIdAllocatorTest, test_get_chunkIds_1001) {
+    EXPECT_CALL(*mockEtcdClient_.get(), Get(_, _))
+        .WillRepeatedly(Invoke(GetRep));
+    EXPECT_CALL(*mockEtcdClient_.get(), CompareAndSwap(_, _, _))
+        .WillRepeatedly(Invoke(CompareAndSwapRep));
+    const int times = 1001;
+    set<uint64_t> chunkids;  // record get chunkids
+    for (int i = 0; i < times; ++i) {
+        uint64_t tmp = 0;
+        chunkIdAllocator_->GenChunkId(&tmp);
+        chunkids.insert(tmp);
+        LOG(INFO) << "allocate id:" << tmp;
+    }
+    LOG(INFO) << "chunkids' szie: " << chunkids.size();
+
+    ASSERT_EQ(times, chunkids.size());
+}
+
+TEST_F(ChunkIdAllocatorTest, test_reclient_get_chunkIds_1001) {
+    // after test_get_chunkIds_1001，client close and restart,
+    // try to get chunkids
+    mockEtcdClient_ = make_shared<MockEtcdClientImpl>();
+    chunkIdAllocator_ = make_shared<ChunkIdAllocatorImpl>(mockEtcdClient_);
```

Comment:
these two lines code is redundant,  for `TEST_F` it will create a test object of `ChunkIdAllocatorTest` and call `SetUp`, then run test body, and call `TearDown`,

Reply:
to check if mds get 1001 chunkIds and crash.
after that recreate chunkIdAllocator_ to get 1001 chunkIds.